### ISSUE TITLE
Unbreakable Obsidian Wall Around Spawn

### ISSUE THREAD

**Issue by Jeffman1993** [escapingnetwork/core-keeper-dedicated#12]
When creating and loading a new world on a server without a graphics card installed, the unbreakable world border wall generates around the central spawn area near the core. Not sure if this is a one off issue for myself or if something changed with an update as I see support was added in a previous commit.

**Comment by xcjs** [comment]
This is related to some of the discussion we had in #7. Make sure you have X11 installed and are mounting the X11 socket.

Without graphics hardware at all, I'm not sure if anything can be done. You can try installing xvfb (a software framebuffer) on the host (not inside the container), but I'm not sure if this will work. You will still need to mount the X11 socket to the container as mentioned.

**Comment by arguser** [comment]
#13 is now merged, please @Jeffman1993 confirm the issue is solved and we'll proceed to close this issue :)

**Comment by Jeffman1993** [comment]
@arguser I gave the new docker image a try. Unfortunately, it hasn't changed the issue on my side as with a new world the wall still generates around the core spawn area. I do however appreciate everyone's effort on the issue. I've been relatively busy lately and can't do much of a deeper dive into the underlying issues at the moment.

**Comment by cjsaylor** [comment]
@arguser @Jeffman1993 The only other thing I can think of that could be causing this might be a race condition because there's no delay between when Xvfb is started and when core keeper is started: https://github.com/escapingnetwork/core-keeper-dedicated/blob/fca3054b8f3a9570b0420f7553e86a3e03de721f/launch.sh#L31

I wonder if we add a small sleep on the launch there if that might resolve it. I'll try to give a try later.

**Comment by cjsaylor** [comment]
That might be why sharing the x11 context from the host machine to the container made it work, because by the time the container launches, the Xvfb process on the host has had ample time to start up.

**Comment by Jeffman1993** [comment]
@cjsaylor Sounds like a good possibility to me. Thanks for prodding at this by the way.

**Comment by arguser** [comment]
@Jeffman1993 keep us posted!

**Comment by cjsaylor** [comment]
Mozilla did a similar thing for their container setup: https://hg.mozilla.org/mozilla-central/rev/922e64883a5b

**Comment by fcumini** [comment]
Is it affecting worlds from games that were started outside the dedicated server?

**Comment by Gymnae** [comment]
Just tried this container on my server without a GPU and ran into the obisdian wall

**Comment by arguser** [comment]
> Just tried this container on my server without a GPU and ran into the obisdian wall

Maybe this suggestion https://github.com/escapingnetwork/core-keeper-dedicated/issues/12#issuecomment-1289391914 solves the situation, going to try to make a PR for it.

**Comment by Micke90s** [comment]
Hallo,

I was facing the same Problem. 
(Linux Debian 12, without display)

I tried the fork of @plveir https://github.com/plveir/core-keeper-dedicated which seems to work (I also installed x11-utils xserver-common on the host before I tried the other docker image, so my test case is not completely clear).

Would it be possible to merge the changes of @plveir to this repository? I think it would be useful if this would be a separate branch which also deploys a separate docker tag. This would avoid that the main branch contains the not required X11 libraries for the most users. If the problem occurs the user only need to change the tag of the docker image. 

Thank you very much for your effort.
